packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -420,12 +420,19 @@ export class ApolloServerBase {
         .catch(err => {
           // We intentionally do not re-throw the exact error from the gateway
           // configuration as it may contain implementation details and this
-          // error will propagate to the client. We will, however, log the error
-          // for observation in the logs.
-          const message = "This data graph is missing a valid configuration.";
-          this.logger.error(message + " " + (err && err.message || err));
+          // error will propagate to the client. We will, however, debug log
+          // the error for observation in the logs. This error is most often
+          // a red herring, which is why we log at a low level.
+          const message = [
+            `Unable to find managed federation configuration for graph ${engineConfig?.graphId}@${engineConfig?.graphVariant}.`,
+            'Please check that this graph has implementing services published to it.',
+            'For more information on setting up managed federation, see here:',
+            'https://www.apollographql.com/docs/studio/managed-federation/setup/.',
+            'To see the original error message, turn on DEBUG level logging.'
+          ].join('\n');
```

Comment:
This error should be thrown at the location where the behavior that this error happens is occuring, not in the root-level catch.